core/src/test/java/com/radixdlt/rev2/LedgerProofsGcTest.java

```diff
@@ -0,0 +1,188 @@
+import com.radixdlt.genesis.GenesisConsensusManagerConfig;
+import com.radixdlt.harness.deterministic.DeterministicTest;
+import com.radixdlt.harness.deterministic.PhysicalNodeConfig;
+import com.radixdlt.harness.predicates.NodePredicate;
+import com.radixdlt.harness.predicates.NodesPredicate;
+import com.radixdlt.harness.simulation.application.TransactionGenerator;
+import com.radixdlt.modules.FunctionalRadixNodeModule;
+import com.radixdlt.modules.FunctionalRadixNodeModule.ConsensusConfig;
+import com.radixdlt.modules.FunctionalRadixNodeModule.LedgerConfig;
+import com.radixdlt.modules.FunctionalRadixNodeModule.NodeStorageConfig;
+import com.radixdlt.modules.FunctionalRadixNodeModule.SafetyRecoveryConfig;
+import com.radixdlt.modules.StateComputerConfig;
+import com.radixdlt.modules.StateComputerConfig.REV2ProposerConfig;
+import com.radixdlt.networks.Network;
+import com.radixdlt.testutil.TestStateReader;
+import com.radixdlt.transactions.RawNotarizedTransaction;
+import com.radixdlt.utils.UInt32;
+import com.radixdlt.utils.UInt64;
+import java.time.Duration;
+import java.util.List;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public final class LedgerProofsGcTest {
+
+  @Rule public TemporaryFolder folder = new TemporaryFolder();
+
+  private DeterministicTest createTest(long mostRecentFullResolutionEpochCount, long epochLength) {
+    return DeterministicTest.builder()
+        .addPhysicalNodes(PhysicalNodeConfig.createBatch(1, true))
+        .messageSelector(firstSelector())
+        .messageMutator(MessageMutator.dropTimeouts())
+        .functionalNodeModule(
+            new FunctionalRadixNodeModule(
+                NodeStorageConfig.tempFolder(folder),
+                true,
+                SafetyRecoveryConfig.MOCKED,
+                ConsensusConfig.of(1000),
+                LedgerConfig.stateComputerNoSync(
+                    new StateComputerConfig.REv2StateComputerConfig(
+                        Network.INTEGRATIONTESTNET.getId(),
+                        GenesisBuilder.createTestGenesisWithNumValidators(
+                            1,
+                            Decimal.ONE,
+                            GenesisConsensusManagerConfig.Builder.testWithRoundsPerEpoch(
+                                epochLength)),
+                        new DatabaseFlags(false, false),
+                        REV2ProposerConfig.transactionGenerator(
+                            new MbTransactionGenerator(NetworkDefinition.INT_TEST_NET), 3),
+                        false,
+                        StateHashTreeGcConfig.forTesting(),
+                        new LedgerProofsGcConfig(
+                            UInt32.fromNonNegativeInt(1),
+                            UInt64.fromNonNegativeLong(mostRecentFullResolutionEpochCount)),
+                        false))));
+  }
+
+  @Test
+  public void node_prunes_non_critical_proofs_from_configured_old_enough_epochs() {
```

Comment:
_(note from author)_

This test takes 14s locally.
It is long because it wants to see that 2 proofs per epoch are left not-pruned. (and to achieve it, each epoch needs to excess the "1000 txns or 12 MB limit")

I am asking since I am not sure about our conventions on test timings.
Is 14s ok? Should I somehow mark it "slow"? Should I somehow make it an "integration" test? Or should I, instead, make the test run ~1s at the price of only testing super-small, single-proof epochs?

Reply:
Maybe we can test it with a lower `MAX_TXNS_FOR_RESPONSES_SPANNING_MORE_THAN_ONE_PROOF` & `MAX_TXN_BYTES_FOR_A_SINGLE_RESPONSE`?

Reply:
Hmm well yes and no, we have a whole section of our configuration for `protocol.*` variables, and sanity checks they're internally consistent. These shouldn't be changed, but are useful to change for (A) investigating what the correct values should be / optimizing and (B) tests.

I think it's reasonable to make these configurable, but have a default which shouldn't be changed in prod. We could even consider requiring `protocol.enable_unsafe_testing_only_protocol_constant_changes` or something to read them :D

Reply:
That said, 14 seconds isn't so bad 👍